TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonToByteBufHandler.java

```diff
@@ -149,28 +143,31 @@ private void writeHeadersIntoByteBufs(ChannelHandlerContext ctx,
 
     private static void writeHeadersAsChunks(ChannelHandlerContext ctx,
                                              HttpJsonMessageWithFaultingPayload httpJson,
-                                             List<Integer> headerChunkSizes,
-                                             int maxLastBufferSize)
-            throws IOException
-    {
-        AtomicInteger chunkIdx = new AtomicInteger(headerChunkSizes.size());
-        var bufs = headerChunkSizes.stream()
-                .map(i -> ctx.alloc().buffer(chunkIdx.decrementAndGet()==0?maxLastBufferSize:i).retain())
-                .toArray(ByteBuf[]::new);
-        CompositeByteBuf cbb = null;
+                                             List<Integer> headerChunkSizes) throws IOException {
+        var initialSize = headerChunkSizes.stream().mapToInt(Integer::intValue).sum();
+
+        ByteBuf buf = null;
         try {
-            cbb = ctx.alloc().compositeBuffer(bufs.length);
-            cbb.addComponents(true, bufs);
-            log.debug("cbb.refcnt=" + cbb.refCnt());
-            try (var bbos = new ByteBufOutputStream(cbb)) {
+            buf = ctx.alloc().buffer(initialSize);
+            try (var bbos = new ByteBufOutputStream(buf)) {
                 writeHeadersIntoStream(httpJson, bbos);
             }
-            for (var bb : bufs) {
-                ctx.fireChannelRead(bb);
+
+            int index = 0;
+            var chunkSizeIterator = headerChunkSizes.iterator();
+            while (index < buf.writerIndex()) {
+                if (!chunkSizeIterator.hasNext()) {
+                    throw Lombok.sneakyThrow(new RuntimeException("Ran out of input chunks for mapping"));
```

Comment:
If this should only happen when there was a programmer error, make the error message a bit more clear.  Use an IllegalStateException too - that will help indicate that it wasn't just a transient runtime problem.

Reply:
Updated to be IllegalStateException